rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -755,6 +750,22 @@ mod tests {
         "value": 1
     }"#;
 
+    const DUMMY_COUNTER_MESSAGE_WITH_SAMPLING_WEIGHT: &str = r#"{
+        "version": 2,
+        "use_case_id": "spans",
+        "org_id": 1,
+        "project_id": 3,
+        "metric_id": 65561,
+        "timestamp": 1704614940,
+        "sentry_received_timestamp": 1704614940,
+        "tags": {"9223372036854776010": "production", "9223372036854776017": "init", "65689": "metric_e2e_spans_counter_v_VUW93LMS"},
+        "retention_days": 90,
+        "mapping_meta":{"h":{"9223372036854776017":"session.status","9223372036854776010":"environment"},"f":{"65689":"metric_e2e_spans_counter_k_VUW93LMS"},"d":{"65561":"c:spans/spans@none"}},
+        "type": "c",
+        "value": 1,
+        "sampling_weight": 100
```

Comment:
Might be a good idea to have the sampling weight be float in one of the messages to test that case.

Reply:
I agree, I changed one of the values to a floating point (`100.1`) and have the test to expect `100` to make sure the cast is working as expected